### PR TITLE
Additional logic to cover non-standard label-to-username mappings

### PR DESCRIPTION
**Pull request by smithki**
Based on user feedback about: https://retail.axisbank.co.in

The above form uses bad HTML semantics, so we're adding a case to try and associated username labels with subsequent input controls.

### CODE REVIEW

File: src/login/field-specs.ts [hunk 1]
```diff
@@ -41,14 +42,31 @@ export const USERNAME_FIELD_SPEC: FieldSpec<HTMLInputElement> = {
...
+          } else {
+            // 3rd case: query for the first subsequent input adjacent to the label
+            const subsequentInputQueries = [`[${labelAttr}="${i}"] ~ input`, `[${labelAttr}="${i}"] ~ * input`];
+            const subsequentInput = label.parentElement?.querySelector(subsequentInputQueries.join(', '));
+            subsequentInput?.setAttribute(usernameAttr, 'yes');
+
+            // If an input cannot be identified by this point, then the website
+            // developer has really messed up their form and we can't really
+            // help it.
+          }
```
Comment (smithki) [inline src/login/field-specs.ts:69]: This is the meat & potatoes right here.

**Comment by dcow** [comment]
Feel free to open a PR upstream if this is generally useful.

**Comment by smithki** [comment]
> Feel free to open a PR upstream if this is generally useful.

We've initiated some larger-scale refactors in the code-base generally that I believe makes it unfit to contribute upstream. I think we're better off keeping our fork separate & iterating progressively towards our specific use-case & need.

**Comment by smithki** [comment]
:rocket: PR was released in [`v0.4.4`](https://github.com/withuno/locust/releases/tag/v0.4.4) :rocket:

**Pull request merged**